services/blockchain-indexer/shared/dataService/business/pos/unlocks.js

```diff
@@ -35,10 +35,21 @@ const getPosUnlocks = async params => {
 		{ address: params.address },
 	);
 
-	// Filter results based on `params.isUnlockable`
-	const filteredPendingUnlocks = 'isUnlockable' in params
-		? pendingUnlocks.filter(u => u.unlockable === params.isUnlockable)
-		: pendingUnlocks;
+	// Filter results based on `params.isLocked`
+	const filteredPendingUnlocks = pendingUnlocks.reduce((accumulator, pendingUnlock) => {
+		const { unlockable, ...remPendingUnlock } = pendingUnlock;
+		const isLocked = !pendingUnlock.unlockable;
+
+		if (params.isLocked === undefined || params.isLocked === isLocked) {
+			accumulator.push({
+				...remPendingUnlock,
+				isLocked,
+			});
+		}
+		return accumulator;
+	},
+	[],
+	);
```

Comment:
Suggested code formatting to clearly highlight the parameter inputs for the `reduce` function based on indentations.

```suggestion
	const filteredPendingUnlocks = pendingUnlocks.reduce(
		(accumulator, pendingUnlock) => {
			const { unlockable, ...remPendingUnlock } = pendingUnlock;
			const isLocked = !pendingUnlock.unlockable;

			// Filter results based on `params.isLocked`
			if (params.isLocked === undefined || params.isLocked === isLocked) {
				accumulator.push({
					...remPendingUnlock,
					isLocked,
				});
			}
			return accumulator;
		},
		[],
	);
```